mkosi/distributions/gentoo.py

```diff
@@ -295,41 +214,64 @@ def mkosi_conf(self) -> None:
             )
         )
 
-    def get_snapshot_of_portage_tree(self) -> None:
-        run_workspace_command(self.state.root, ["/usr/bin/emerge-webrsync"], network=True,
-                              env=self.state.environment)
-
-    def update_stage3(self) -> None:
-        invoke_emerge(self.state, opts=self.EMERGE_UPDATE_OPTS, pkgs=self.pkgs['boot'])
-        invoke_emerge(self.state, opts=["--config"], pkgs=["sys-kernel/gentoo-kernel-bin"])
-        invoke_emerge(self.state, opts=self.EMERGE_UPDATE_OPTS, pkgs=self.pkgs['sys'])
+        repos_cfg = cls.stage3_cache / BINREPOS_CONF_FILE
+        with repos_cfg.open(mode='a') as f:
+            for repo in state.config.repositories:
+                f.write(
+                    dedent(
+                        f"""\
+                        # MKOSI
+                        [binhost]
+                        sync-uri = {repo}
+                        """
+                    )
+                )
+
+        root_portage_cfg = state.root
+        root_portage_cfg /= user_config_path.relative_to(cls.stage3_cache)
+        root_portage_cfg.mkdir(parents=True, exist_ok=True)
+        copy_path(user_config_path, root_portage_cfg)
+
+        bwrap_params: list[PathString] = [
+            "--bind", state.cache_dir / "repos", "/var/db/repos"
+        ]
+        run_workspace_command(cls.stage3_cache, ["/usr/bin/emerge-webrsync"],
+                              bwrap_params=bwrap_params, network=True)
+
+        opts = [
+            "--with-bdeps=n",
+            "--complete-graph-if-new-use=y",
+            "--verbose-conflicts",
+            "--changed-use",
+            "--newuse",
+            "--root-deps=rdeps",
+        ]
+        with complete_step("merging stage2"):
+            invoke_emerge(state, sysroot=cls.stage3_cache,
+                          opts=opts+["--emptytree", "--nodeps"],
+                          pkgs=["sys-apps/baselayout"],
+                          env={**emerge_vars, 'USE': 'build'})
 
-    def depclean(self) -> None:
-        invoke_emerge(self.state, actions=["--depclean"])
+        opts += ["--noreplace"]
 
+        with complete_step("Merging bare minimal atoms"):
+            invoke_emerge(state, sysroot=cls.stage3_cache,
+                          opts=opts+["--exclude", "sys-devel/*"],
+                          pkgs=["app-shells/bash", "sys-apps/systemd"],
+                          env=emerge_vars)
```

Comment:
Can we drop this? Ideally the base system without any configuration has just the base directory layout and /etc/passwd + related files, nothing else